.github/workflows/eslint.yml

```diff
@@ -0,0 +1,54 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+# ESLint is a tool for identifying and reporting on patterns
+# found in ECMAScript/JavaScript code.
+# More details at https://github.com/eslint/eslint
+# and https://eslint.org
+
+name: ESLint
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+
+jobs:
+  eslint:
+    name: Run eslint scanning
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      security-events: write
+      actions: read # only required for a private repository by github/codeql-action/upload-sarif to get the Action run status
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+          # We must fetch at least the immediate parents so that if this is
+          # a pull request then we can checkout the head.
+        fetch-depth: 2
+
+      # If this run was triggered by a pull request event, then checkout
+      # the head of the pull request instead of the merge commit.
+      - run: git checkout HEAD^2
+        if: ${{ github.event_name == 'pull_request' }}
+
+      - name: Install ESLint
+        run: |
+          npm install eslint@8.10.0
+          npm install @microsoft/eslint-formatter-sarif@2.1.7
+
+      - name: Run ESLint
+        run: npx eslint .
+          --config .eslintrc.js
+          --ext .js,.jsx,.ts,.tsx
```

Comment:
there is already `test_eslint` action in our `package.json`, I think we can use it instead of passing `--config` and `--ext` parameters here:
```sh
npm run test_eslint
```

Reply:
Hi @sosnovsky. We may use that but after a few more thinking, I think its best to write the custom test within the GitHub action so all the ESLint test related and run by the GitHub action is in one place. Not unlike updating the test_eslint or even wrote another specifically for the ESLint GitHub action then passing some more addition flags like --output-file and --format...

Meanwhile, my fixes here just temporary since I found a minor bug which needs to be fixed within the "@microsoft/eslint-formatter-sarif" npm package from which the duplicates json items are not omitted causing issues when uploading the sarif file.

[876c442](https://github.com/FlowCrypt/flowcrypt-browser/pull/5735/commits/876c44221a819809105f32fc06321ca7fda2b901)

Luckly, eslint's --format flag accepts custom formatter file and so I have to commit the customized file of the eslint-formatter-sarif in https://github.com/microsoft/sarif-js-sdk/blob/main/packages/eslint-formatter-sarif/sarif.js.

Finally,  confirmation that this works would be the comment here from "[github-advanced-security](https://github.com/apps/github-advanced-security)".

Cheers,